app/browser/tabs.js

```diff
@@ -536,6 +536,9 @@ const api = {
         createProperties.parent_partition = ''
       }
     }
+    if (createProperties.isPrivate && createProperties.partition === undefined) {
```

Comment:
Pls move this into `getPartition` which takes in createProperties and returns the partition.